Editor/Resources/EditorWindow/GameLiftPlugin.cs

```diff
@@ -0,0 +1,155 @@
+        private const string TabButtonClassName = "TabButton";
+        private const string TabContentClassName = "TabContent";
+
+        private static GameLiftPlugin GetWindow()
+        {
+            var inspectorType = Type.GetType("UnityEditor.GameView,UnityEditor.dll");
+            var window = GetWindow<GameLiftPlugin>(inspectorType);
+            window.titleContent = new GUIContent("Amazon GameLift", window.icon);
+            return window;
+        }
+
+        [MenuItem("Amazon GameLift/Show Amazon GameLift Window", priority = 0)]
+        public static void ShowWindow()
+        {
+            GetWindow();
+        }
+
+        [MenuItem("Amazon GameLift/Bring Panel to Front", priority = 1)]
+        public static void FocusPanel()
+        {
+            ShowWindow();
+        }
+
+        [MenuItem("Amazon GameLift/Set AWS Account Profiles", priority = 100)]
+        public static void OpenAccountProfilesTab()
+        {
+            GetWindow().OpenTab("Credentials");
+        }
+
+        [MenuItem("Amazon GameLift/Host with Anywhere", priority = 101)]
+        public static void OpenAnywhereTab()
+        {
+            GetWindow().OpenTab("Anywhere");
+        }
+
+        [MenuItem("Amazon GameLift/Host with Managed EC2", priority = 102)]
+        public static void OpenEC2Tab()
+        {
+            GetWindow().OpenTab("EC2");
+        }
+
+        [MenuItem("Amazon GameLift/Import Sample Game", priority = 103)]
+        public static void ImportSampleGame()
+        {
+            string filePackagePath = $"Packages/{Paths.PackageName}/{Paths.SampleGameInPackage}";
+            AssetDatabase.ImportPackage(filePackagePath, interactive: true);
+        }
+
+        [MenuItem("Amazon GameLift/Help/Documentation", priority = 200)]
+        public static void OpenDocumentation()
+        {
+            Application.OpenURL(Urls.AwsHelpGameLiftUnity);
+        }
+
+        [MenuItem("Amazon GameLift/Help/AWS GameTech Forum", priority = 201)]
+        public static void OpenGameTechForums()
+        {
+            Application.OpenURL(Urls.AwsGameTechForums);
+        }
+
+        [MenuItem("Amazon GameLift/Help/Report Issues", priority = 202)]
+        public static void OpenReportIssues()
+        {
+            Application.OpenURL(Urls.GitHubAwsLabs);
+        }
+
+        private void CreateGUI()
+        {
+            _root = rootVisualElement;
+            _mVisualTreeAsset = UnityEngine.Resources.Load<VisualTreeAsset>("EditorWindow/GameLiftPlugin");
+            if (_mVisualTreeAsset == null)
+            {
+                return;
+            }
+
+            VisualElement uxml = _mVisualTreeAsset.Instantiate();
+            _root.Add(uxml);
+            
+            ApplyText();
+
+            var contentContainer = _root.Q(className: "main__content").Children().First();
+            var landingPage = new LandingPage(contentContainer);
```

Comment:
We appear to be inferring how the LandingPage is structured when we pass the contentContainer in order to correctly open tabs. We should avoid this as it's brittle. I would recommend the main plugin has full control over the elements it's accessing

Reply:
Do you mean the fact that LandingPage has the name & class which control its display? 

If so, would we prefer to create a container for each page in the main plugin file + pass those to the individual pages, e.g.
```html
...
<ui:ScrollView name="Content" class="main__content">
    <ui:VisualElement name="LandingTab" class="TabContent" />
....
```
```cs
var landingContainer = _root.Q("LandingTab");
var landingPage = new LandingPage(landingContainer);
```
Or we could have the Pages have their root element public, and set up the required attributes directly on that, e.g.

```cs
// LandingPage.cs
public class LandingPage
{
  public readonly VisualElement Root;
  
  public LandingPage(VisualElement container)
  {
    var mVisualTreeAsset = UnityEngine.Resources.Load<VisualTreeAsset>("EditorWindow/Pages/LandingPage");
    Root = mVisualTreeAsset.Instantiate();

    container.Add(Root);
  
// GameLiftPlugin.cs
private void SetupTab(VisualElement tabContent, string name)
{
  tabContent.name = $"{name}Content";
  tabContent.AddToClassList(TabContentClassName);
}
...
  var contentContainer = _root.Q(className: "main__content");
  var landingPage = new LandingPage(landingContainer);
  SetupTab(landingPage, Pages.Landing);
```

Reply:
The first suggestion is what I was thinking myself, but I do like the second option except that we're modifying the name and class in the `LandingPage.uxml`. I'd prefer that `GameLiftPlugin.*` and `LandingPage.*` are isolated to their own logic. Consider a small tweak to create the container for the landing page to sit in as part of the tab setup:
```
// GameLiftPlugin.cs
private void SetupTab(VisualElement tabContent, string name)
{
  tabContent.name = $"{name}Content";
  tabContent.AddToClassList(TabContentClassName);
}
...
  var contentContainer = _root.Q(className: "main__content");
  var landingContent = new VisualElement();
- var landingPage = new LandingPage(contentContainer);
- SetupTab(landingPage, Pages.Landing);
+ var landingPage = new LandingPage(landingContent);
+ SetupTab(landingContent, Pages.Landing);
```

Reply:
I have address this by having `SetupTab` take a name for the tab, and it created and attachs a container to the parent. It then returns the container it created so we can pass it straight into the Page constructor